integrationtests/tests/fixtures/mod.rs

```diff
@@ -149,13 +150,25 @@ pub async fn fixtures(num_peers: u16) -> anyhow::Result<Fixtures> {
 
     // in case we need to output logs using 'cargo test -- --nocapture'
     if base_port == BASE_PORT_INIT {
-        tracing_subscriber::fmt()
-            .with_env_filter(
-                EnvFilter::try_from_default_env()
-                    .unwrap_or_else(|_| EnvFilter::new("info,fedimint::consensus=warn")),
-            )
-            .init();
+        if env::var_os("FEDIMINT_TRACE_CHROME").map_or(false, |x| !x.is_empty()) {
+            let (cr_layer, gaurd) = tracing_chrome::ChromeLayerBuilder::new()
+                .include_args(true)
+                .build();
+            // drop gaurd cause file to written and closed
+            // in this case file will closed after exit of program
+            std::mem::forget(gaurd);
```

Comment:
How does it work? Without a guard the rest of the code there doesn't know that we're exiting? It might be that it mostly works, but it can truncate the output.

IMO, you should return `Option<Guard>` from both sides here, `Some` from here, `None` from the other one.

Reply:
Oh. Given that we run all tests in parallel, I'm not even sure how useful this will be with all tracing mixed together.

A thread_local holding a `Weak<Guard>` that tests acquire and upgrade to `Arc<Guard>` (and initialize if empty) would probably do the right thing, though the problem of inter-mixing logging would still be there.

Reply:
Then people can just filter the test they are interested in, and/or run them serially.

Reply:
Yes, it is only useful in single test mode. But then `Drop` for thread local doesn't run for main thread on linux :/

Reply:
That's why we use `Weak` in the thread local - so it doesn't hold any references and everything drops after last test failed.

Reply:
Do you think @dpc's approach is viable? Sounds a bit cleaner tbh. Otoh I don't mind this hack in test code :laughing: Can not running `drop` cause problems in any way? Like for example not flushing buffers?